tests/NHSD.BuyingCatalogue.Solutions.API.UnitTests/NativeDesktop/NativeDesktopMemoryAndStorageControllerTests.cs

```diff
@@ -0,0 +1,110 @@
+    {
+        [TestFixture]
+        public sealed class NativeDesktopThirdPartyControllerTests
+        {
+            private Mock<IMediator> _mediatorMock;
+            private NativeDesktopMemoryAndStorageController _nativeDesktopMemoryAndStorageController;
+            private readonly string _solutionId = "Sln1";
+            private Mock<ISimpleResult> _simpleResultMock;
+            private Dictionary<string, string> _resultDictionary;
+
+            [SetUp]
+            public void Setup()
+            {
+                _mediatorMock = new Mock<IMediator>();
+                _nativeDesktopMemoryAndStorageController = new NativeDesktopMemoryAndStorageController(_mediatorMock.Object);
+                _simpleResultMock = new Mock<ISimpleResult>();
+                _simpleResultMock.Setup(x => x.IsValid).Returns(() => !_resultDictionary.Any());
+                _simpleResultMock.Setup(x => x.ToDictionary()).Returns(() => _resultDictionary);
+                _resultDictionary = new Dictionary<string, string>();
+                _mediatorMock.Setup(x =>
+                        x.Send(It.IsAny<UpdateNativeDesktopMemoryAndStorageCommand>(),
+                            It.IsAny<CancellationToken>()))
+                    .ReturnsAsync(() => _simpleResultMock.Object);
+            }
+
+            [Test]
+            public async Task UpdateValidMemoryAndStorageDetails()
+            {
+                var request = new UpdateNativeDesktopMemoryAndStorageViewModel
+                {
+                    MinimumMemoryRequirement = "1 bit",
+                    StorageRequirementsDescription = "A description",
+                    MinimumCpu = "1Hz",
+                    RecommendedResolution = "1000,00000,0000x1"
+                };
+
+                var result =
+                    (await _nativeDesktopMemoryAndStorageController.Update(_solutionId, request)
+                        .ConfigureAwait(false)) as NoContentResult;
+                result.StatusCode.Should().Be((int)HttpStatusCode.NoContent);
+                _mediatorMock.Verify(
+                    x => x.Send(
+                        It.Is<UpdateNativeDesktopMemoryAndStorageCommand>(c =>
+                            c.SolutionId == _solutionId &&
+                            c.Data.MinimumMemoryRequirement == request.MinimumMemoryRequirement &&
+                            c.Data.StorageRequirementsDescription == request.StorageRequirementsDescription &&
+                            c.Data.MinimumCpu == request.MinimumCpu &&
+                            c.Data.RecommendedResolution == request.RecommendedResolution
+                        ),
+                        It.IsAny<CancellationToken>()), Times.Once);
+            }
+
+            [Test]
+            public async Task UpdateInvalidReturnsBadRequestWithValidationDetails()
+            {
+                _resultDictionary.Add("minimum-memory-requirement", "required");
```

Comment:
Is it worth showing all cases of the validation in the controller tests?

Reply:
I don't think so, all we're verifying is that whatever is returned by Mediatr is reflected accurately in the returned result. The cases are covered by the Handler tests